builtin/aws/lambda/platform.go

```diff
@@ -271,11 +279,22 @@ func (p *Platform) Deploy(
 		}
 
 		funcCfg, err := lamSvc.UpdateFunctionCode(&lambda.UpdateFunctionCodeInput{
-			FunctionName: aws.String(src.App),
-			ImageUri:     aws.String(img.Name()),
+			FunctionName:  aws.String(src.App),
+			ImageUri:      aws.String(img.Name()),
+			Architectures: aws.StringSlice([]string{architecture}),
 		})
 
 		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case "ValidationException":
+					// likely here if Architectures was invalid
+					if architecture != lambda.ArchitectureX8664 && architecture != lambda.ArchitectureArm64 {
+						return nil, fmt.Errorf("architecture must be either x86_64 or arm64")
```

Comment:
```suggestion
						return nil, fmt.Errorf("architecture must be either \"x86_64\" or \"arm64\"")
```

Small nit, but suggested values here are easier to read I think if we quote them.